path/base.go

```diff
@@ -0,0 +1,77 @@
+package path
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+const (
+	notation = "notation"
+)
+
+var (
+	// for mocking
+	goos          = runtime.GOOS
+	userConfigDir = os.UserConfigDir
+	userCacheDir  = os.UserCacheDir
+	getenv        = os.Getenv
+
+	// SystemLibexec directory for binaries not meant to be executed directly by users' shell or scripts
+	SystemLibexec string
+	// SystemConfig directory for configurations
+	SystemConfig string
+
+	// UserLibexec is user level libexec directory
+	UserLibexec string
+	// UserConfig is user level config directory
+	UserConfig string
+	// UserCache for user-specific cache
+	UserCache string
+)
+
+func init() {
+	loadPath()
+}
+
+// init function defines the directory for NotationLibexec, NotationConfig, NotationCache
+func loadPath() {
+	var err error
+	// set system config and libexec
+	switch goos {
+	case "darwin":
+		SystemConfig = "/Library/Application Support/notation"
+		SystemLibexec = "/usr/local/lib/notation"
+	case "windows":
+		SystemConfig = getenv("ProgramData")
+		if SystemConfig == "" {
+			panic("environment variable `ProgramData` is not set.")
```

Comment:
Would this show an ugly message with stacktrace to notation CLI users like [this](https://go.dev/play/p/mlYR1fCZ7L1?v=goprev)

We may want to present the error in a better way

Reply:
1. this environment variable must exist, or the OS is not supported. I prefer to use panic, because this case is really need to stop the program, also the LoadPath() function should be called before any other function in this package, that is to say LoadPath() should be called in init() and return error is no use.
2. Yes it is ugly. I think we should call recover() in the main() to handle it, which should be in the notation-cli.